toolkit.nu

```diff
@@ -34,27 +34,19 @@ export def fmt [
 export def clippy [
     --verbose # print extra information about the command's progress
     --features: list<string> # the list of features to run *Clippy* on
-    --workspace # run the *Clippy* command on the whole workspace (overrides `--features`)
 ] {
     if $verbose {
         print $"running ('toolkit clippy' | pretty-format-command)"
     }
 
-    try {
-        if $workspace {(
-            cargo clippy
-                --workspace
-            --
-                -D warnings
-                -D clippy::unwrap_used
-        )} else {(
-            cargo clippy
-                --features ($features | str join ",")
-            --
-                -D warnings
-                -D clippy::unwrap_used
-        )}
-    } catch {
+    try {(
+        cargo clippy
+            --workspace
+            --tests
+            --features ($features | str join ",")
```

Comment:
an old `cargo` incomprehension from my side... :scream: :laughing: 

i still do not understand what would `--features` do when `--workspace` is used :thinking: 
all the features are in the workspace so running with `--workspace` will apply to all the `--features`, right?

a while back i did run test for that and was not convinced by the use of `--features` when `--workspace` is already there, this is why i wrote this command like that :yum:

Reply:
answered in https://github.com/nushell/nushell/issues/10528#issuecomment-1741274893